### PR TITLE
Initialize RNGs for trainers differently by default

### PR DESCRIPTION
**Pull request by JaeseungYeom**
Basically change the interface ```random_init_models_differently: true``` to ```random_init_trainers_identically: false```

Passed baboo tests https://lc.llnl.gov/bamboo/browse/LBANN-JAES35

### CODE REVIEW

File: src/proto/trainer.proto [hunk 1]
```diff
@@ -49,6 +49,10 @@ message Trainer {
   // background.
   int64 num_parallel_readers = 3;
 
+
+  // If false, trainers will have their trainer rank mixed into their random seed.
+  bool random_init_trainers_identically = 4;
```
Comment (timmoon10) [inline src/proto/trainer.proto]: Can we put this under the "Advanced options"?

File: src/utils/lbann_library.cpp [hunk 2]
```diff
@@ -182,16 +182,16 @@ std::unique_ptr<model> build_model_from_prototext(
 
   // Initialize models differently if needed.
 #ifndef LBANN_DETERMINISTIC
-  if (pb_model->random_init_models_differently()) {
+  if (!pb_trainer->random_init_trainers_identically()) {
     random_seed = random_seed + comm->get_trainer_rank();
```
Comment (timmoon10) [inline src/utils/lbann_library.cpp]: Perhaps beyond the scope of this PR, but we have a [utility function to mix RNG seeds](https://github.com/LLNL/lbann/blob/279b4d4a5f0268519c9ce91f5940a59a355d1336/include/lbann/utils/hash.hpp#L47):
```suggestion
     random_seed = hash_combine(random_seed, comm->get_trainer_rank());
```

File: src/proto/trainer.proto (hunk 1 above)
Comment (JaeseungYeom) [reply src/proto/trainer.proto]: sure

File: src/utils/lbann_library.cpp (hunk 2 above)
Comment (JaeseungYeom) [reply src/utils/lbann_library.cpp]: I will apply this.

**Comment by JaeseungYeom** [comment]
Should I merge?

**Pull request merged**